pandas/core/indexes/base.py

```diff
@@ -5032,7 +5033,11 @@ def _get_engine_target(self) -> ArrayLike:
         Get the ndarray or ExtensionArray that we can pass to the IndexEngine
         constructor.
         """
-        return self._values
+        vals = self._values
+        if isinstance(vals, StringArray):
```

Comment:
is this true for other EAs? e.g. original report was for Int64

Reply:
#45652 was about string dtype.  I think Joris commented about Int64 elsewhere and pointed out they are related.